modules/ln_glcore/glcore.scm

```diff
@@ -181,50 +220,246 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
       )))
 
 ;; ----------------------------------
-;; textures 
-
-;; each entry is a vector of initflag,texure,w,h,u8data,pixeltype
-(define glCore:textures (##still-copy (make-table)))
-(define glCore:tidx 0)
-(define glCore:curtexture -1)
-
-(define (glCoreTextureCreate w h data . aux)
-  (glcore:log 5 "glCoreTextureCreate")
-  (let* ((o1x (pair? aux))
-         (o2 (and o1x (cdr aux))))
-    (let ((idx glCore:tidx)
-          (pixeltype
-            (cond
+;; textures
+
+(cond-expand ;; CONSTRUCTION-CASE
+ ((or gambit debug) ;; tentative changes
+  ;;; intentions:
+  ;;; 1. hide globals glCore:textures and glCore:tidx (at least)
+  ;;; 2. (short term) replace vector with distinct type
+
+  ;; (: (%%glCore:textures-ref t d) <<== (table-ref [abstract:glCore:textures] t d))
+  (define %%glCore:textures-ref)
+  ;; glCoreTextureCreate EXPORTED - ubiquitious
+  ;;;
+  ;;; (: (glCoreTextureCreate w h data #!optional (interpolation GL_LINEAR) (wrap GL_CLAMP))
+  ;;;    -> fixnum)
+  (define glCoreTextureCreate)
+  ;; glCoreTextureReset -- TBD: unknown usage status
+  ;;;
+  ;;; (: glCoreTextureReset -> undefined)
+  ;;;
+  ;;; purpose: clear resources
+  (define glCoreTextureReset)
+
+  ;; Implementation (volatile)
+
+  (define-type glCore:texture
+    macros: prefix: %MATURITY+3%texture%macro-
+    %%valid ;; FIXME: factor out from immutable components
+    glidx   ;; index (for opengl and internal table)
+    %%-???-u32vector ;; what is this? mutable?
```

Comment:
what are %%-??? functions - I don't understand the macro piece to adjudicate this so we can't merge it as I can't maintain it.